core/src/main/java/org/mskcc/cbio/portal/servlet/CommonQueryBuilder.java

```diff
@@ -0,0 +1,625 @@
+			accessControl = SpringUtil.getAccessControl();
+		} catch (PolicyException e) {
+			throw new ServletException(e);
+		}
+	}
+
+	/**
+	 * Handles HTTP GET Request.
+	 *
+	 * @param httpServletRequest
+	 *            Http Servlet Request Object.
+	 * @param httpServletResponse
+	 *            Http Servlet Response Object.
+	 * @throws ServletException
+	 *             Servlet Error.
+	 * @throws IOException
+	 *             IO Error.
+	 */
+	protected void doGet(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse)
+			throws ServletException, IOException {
+		doPost(httpServletRequest, httpServletResponse);
+	}
+
+	/**
+	 * Handles HTTP POST Request.
+	 *
+	 * @param httpServletRequest
+	 *            Http Servlet Request Object.
+	 * @param httpServletResponse
+	 *            Http Servlet Response Object.
+	 * @throws ServletException
+	 *             Servlet Error.
+	 * @throws IOException
+	 *             IO Error.
+	 */
+	protected void doPost(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse)
+			throws ServletException, IOException {
+		XDebug xdebug = new XDebug(httpServletRequest);
+		xdebug.startTimer();
+
+		xdebug.logMsg(this, "Attempting to initiate new user query.");
+		Integer dataTypePriority = null;
+		// boolean isVirtualStudy = false;
+
+		String[] cancerStudyIdList = null;
+		// Get User Selected Genetic Profiles
+		HashSet<String> geneticProfileIdSet = new HashSet<String>();
+		String sampleSetId = null;
+		String sampleIds = "";
+		ArrayList<GeneticProfile> profileList = new ArrayList<GeneticProfile>();
+		ArrayList<SampleList> sampleSetList = new ArrayList<SampleList>();
+
+		if (httpServletRequest.getRequestURL() != null) {
+			httpServletRequest.setAttribute(ATTRIBUTE_URL_BEFORE_FORWARDING,
+					httpServletRequest.getRequestURL().toString());
+		}
+
+		// Get User Selected Action
+		String action = httpServletRequest.getParameter(ACTION_NAME);
+
+		String cancerStudyIdListString = httpServletRequest.getParameter(CANCER_STUDY_LIST);
+		String geneList = httpServletRequest.getParameter(GENE_LIST);
+		if (httpServletRequest instanceof XssRequestWrapper) {
+			geneList = ((XssRequestWrapper) httpServletRequest).getRawParameter(GENE_LIST);
+		}
+		geneList = servletXssUtil.getCleanInput(geneList);
+		httpServletRequest.setAttribute(GENE_LIST, geneList);
+		try {
+			List<CancerStudy> cancerStudyList = accessControl.getCancerStudies();
+
+			if (cancerStudyIdListString == null) {
+				String cancerTypeId = httpServletRequest.getParameter(CANCER_STUDY_ID);
+				if(cancerTypeId == null){
+					cancerStudyIdList = new String[] { cancerStudyList.get(0).getCancerStudyStableId() };
+				}else{
+					cancerStudyIdList = new String[] { cancerStudyList.get(0).getCancerStudyStableId() };
+				}
+			} else {
+				cancerStudyIdList = cancerStudyIdListString.split(",");
+			}
+
+			httpServletRequest.setAttribute(CANCER_STUDY_ID, cancerStudyIdList[0]);
```

Comment:
Most of the flows(in jsp's) require 'CANCER_STUDY_ID' and I don't want break existing flows for single study queries. Once we implement all tabs to support both single and multi study then we can get rid of this attribute.